pocket_lab/pl_gui_views.c

```diff
@@ -924,26 +951,38 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
 
 				/* Remove previously enabled channels from pl_gui_capture_chart_ovrly series */
 				for (cnt = 0; cnt < pl_gui_capture_chn_cnt; cnt++) {
-					if (lv_obj_get_state(pl_gui_capture_chn_checkbox[cnt]) == LV_STATE_CHECKED) {
+					if (lv_obj_get_state(pl_gui_capture_chn_checkbox[cnt]) ==
+					    (LV_STATE_CHECKED | LV_STATE_DISABLED)) {
+						lv_obj_clear_state(pl_gui_capture_chn_checkbox[cnt],
+								   LV_STATE_CHECKED| LV_STATE_DISABLED);
 						lv_chart_remove_series(pl_gui_capture_chart_ovrly, pl_gui_capture_chn_ser[cnt]);
 						pl_gui_capture_chn_ser[cnt] = NULL;
+					} else {
+						lv_obj_clear_state(pl_gui_capture_chn_checkbox[cnt],LV_STATE_DISABLED);
 					}
 				}
+				pl_gui_capture_is_running = !pl_gui_capture_is_running;
 			} else {
 				lv_label_set_text(label, "Stop");
 				lv_obj_set_style_bg_color(btn, lv_palette_main(LV_PALETTE_RED), LV_PART_MAIN);
 
 				/* Add enabled channels to pl_gui_capture_chart_ovrly series */
 				for (cnt = 0; cnt < pl_gui_capture_chn_cnt; cnt++) {
 					if (lv_obj_get_state(pl_gui_capture_chn_checkbox[cnt]) == LV_STATE_CHECKED) {
-						pl_gui_capture_chn_ser[cnt] = lv_chart_add_series(pl_gui_capture_chart_ovrly,
-									      lv_palette_main(pl_gui_capture_chn_ser_col[cnt]),
-									      LV_CHART_AXIS_PRIMARY_Y);
-						lv_chart_set_point_count(pl_gui_capture_chart_ovrly, 400);
+						num_of_enabled_channels++;
+						lv_obj_add_state(pl_gui_capture_chn_checkbox[cnt],
+								 LV_STATE_CHECKED | LV_STATE_DISABLED);
+					} else {
+						lv_obj_add_state(pl_gui_capture_chn_checkbox[cnt], LV_STATE_DISABLED);
 					}
+					pl_gui_capture_chn_ser[cnt] = lv_chart_add_series(pl_gui_capture_chart_ovrly,
+								      lv_palette_main(pl_gui_capture_chn_ser_col[cnt]),
+								      LV_CHART_AXIS_PRIMARY_Y);
+					lv_chart_set_point_count(pl_gui_capture_chart_ovrly, 400);
 				}
+				if (num_of_enabled_channels)
```

Comment:
Brackets are missing

Reply:
Apologies, have missed on this. Will update